lib/craft/clock_bound/client.ex

```diff
@@ -0,0 +1,233 @@
+defmodule ClockBound.Client do
+  @moduledoc """
+  This module provides functionality to retrieve clock error bounds and clock status from shared memory.
+  Reads ClockBound data from shared memory generated by the clockbound daemon.
+
+  ## Features
+
+  - Reads binary data from a shared memory file and parses it according to the ClockBound protocol specification.
+  - Provides the current clock error bounds (`earliest` and `latest`) and clock status.
+  - Converts timestamps to `DateTime` for easier comparision.
+
+  ## Usage
+
+  1. Call `ClockBound.Client.now/0` or `ClockBound.Client.now/1` to retrieve the current clock error bounds and status:
+    ```elixir
+    case ClockBound.Client.now() do
+      {:ok, clock_data} ->
+        IO.inspect(clock_data)
+
+      {:error, reason} ->
+        _
+    end
+    ```
+  2. Call `ClockBound.Client.compare/2` to compare two ClockBound windows:
+    ```elixir
+    case ClockBound.Client.compare(clock_data_1, clock_data_2) do
+      :lt -> IO.puts("Clock 1 is less than Clock 2")
+      :gt -> IO.puts("Clock 1 is greater than Clock 2")
+      :ov -> IO.puts("Clock 1 and Clock 2 overlap")
+    end
+    ```
+
+  ## Notes
+
+  - The default shared memory path is `/var/run/clockbound/shm`. You can override this by passing a custom path to the functions.
+  - The module adheres to the ClockBound protocol specification, which can be found [here](https://github.com/aws/clock-bound/blob/main/docs/PROTOCOL.md).
+  """
+
+  @default_shm_path "/var/run/clockbound/shm"
+  @clockbound_restart_grace_period 5 * 1_000_000_000
+
+  # 'AMZN CB'
+  @magic_0 0x414D5A4E
+  @magic_1 0x43420200
+
+  defstruct [
+    :earliest,
+    :latest,
+    :clock_status
+  ]
+
+  @type clock_status :: :unknown | :synchronized | :free_running | :invalid
+  @type t :: %__MODULE__{
+          earliest: DateTime.t(),
+          latest: DateTime.t(),
+          clock_status: clock_status()
+        }
+
+  @doc """
+  Returns the current clock time with error bounds and clock status.
+  """
+  def now(shm_path \\ @default_shm_path) do
+    with {:ok, data} <- read(shm_path),
+         real = :os.system_time(:nanosecond),
+         monotonic = monotonic_time(),
+         {:ok, {earliest, latest, clock_status}} <- compute_bound_at(real, monotonic, data),
+         {:ok, earliest} <- DateTime.from_unix(earliest, :nanosecond),
+         {:ok, latest} <- DateTime.from_unix(latest, :nanosecond) do
+      {:ok,
+       %__MODULE__{
+         earliest: earliest,
+         latest: latest,
+         clock_status: clock_status
+       }}
+    else
+      {:error, reason} -> {:error, reason}
+      _ -> {:error, :unknown}
```

Comment:
it's not unknown, it's just unexpected. :)

```suggestion
      error -> {:error, error}
```